src/script/opcodes/constant.zig

```diff
@@ -0,0 +1,309 @@
+            .OP_NUMEQUAL => "OP_NUMEQUAL",
+            .OP_NUMEQUALVERIFY => "OP_NUMEQUALVERIFY",
+            .OP_NUMNOTEQUAL => "OP_NUMNOTEQUAL",
+            .OP_LESSTHAN => "OP_LESSTHAN",
+            .OP_GREATERTHAN => "OP_GREATERTHAN",
+            .OP_LESSTHANOREQUAL => "OP_LESSTHANOREQUAL",
+            .OP_GREATERTHANOREQUAL => "OP_GREATERTHANOREQUAL",
+            .OP_MIN => "OP_MIN",
+            .OP_MAX => "OP_MAX",
+            .OP_WITHIN => "OP_WITHIN",
+            .OP_RIPEMD160 => "OP_RIPEMD160",
+            .OP_SHA1 => "OP_SHA1",
+            .OP_SHA256 => "OP_SHA256",
+            .OP_HASH160 => "OP_HASH160",
+            .OP_HASH256 => "OP_HASH256",
+            .OP_CODESEPARATOR => "OP_CODESEPARATOR",
+            .OP_CHECKSIG => "OP_CHECKSIG",
+            .OP_CHECKSIGVERIFY => "OP_CHECKSIGVERIFY",
+            .OP_CHECKMULTISIG => "OP_CHECKMULTISIG",
+            .OP_CHECKMULTISIGVERIFY => "OP_CHECKMULTISIGVERIFY",
+            .OP_NOP1 => "OP_NOP1",
+            .OP_CHECKLOCKTIMEVERIFY, .OP_NOP2 => "OP_CHECKLOCKTIMEVERIFY",
+            .OP_CHECKSEQUENCEVERIFY, .OP_NOP3 => "OP_CHECKSEQUENCEVERIFY",
+            .OP_NOP4 => "OP_NOP4",
+            .OP_NOP5 => "OP_NOP5",
+            .OP_NOP6 => "OP_NOP6",
+            .OP_NOP7 => "OP_NOP7",
+            .OP_NOP8 => "OP_NOP8",
+            .OP_NOP9 => "OP_NOP9",
+            .OP_NOP10 => "OP_NOP10",
+            .OP_PUBKEYHASH => "OP_PUBKEYHASH",
+            .OP_PUBKEY => "OP_PUBKEY",
+            .OP_INVALIDOPCODE => "OP_INVALIDOPCODE",
+        };
+    }
+
+    pub fn isDisabled(self: Opcode) bool {
+        return switch (self) {
+            .OP_VER, .OP_VERIF, .OP_VERNOTIF, .OP_2MUL, .OP_2DIV => true,
+            else => false,
+        };
+    }
+
+    pub fn isReserved(self: Opcode) bool {
+        return switch (self) {
+            .OP_RESERVED, .OP_RESERVED1, .OP_RESERVED2 => true,
+            else => false,
+        };
+    }
+
+    pub fn isPushData(self: Opcode) bool {
+        return switch (self) {
+            .OP_PUSHDATA1, .OP_PUSHDATA2, .OP_PUSHDATA4 => true,
+            else => false,
+        };
+    }
+
+    pub fn isConditional(self: Opcode) bool {
+        return switch (self) {
+            .OP_IF, .OP_NOTIF, .OP_ELSE, .OP_ENDIF => true,
+            else => false,
+        };
+    }
+
+    pub fn fromByte(byte: u8) !Opcode {
+        return std.meta.intToEnum(Opcode, byte) catch {
+            return EngineError.UnknownOpcode;
+        };
+    }
+};
+
+// Helper function to get all opcodes
+pub fn getAllOpcodes() []const Opcode {
+    return std.enums.values(Opcode);
+}
+
+// Helper function to get opcode from string representation
+pub fn opcodeFromString(name: []const u8) ?Opcode {
+    for (getAllOpcodes()) |opcode| {
+        if (std.mem.eql(u8, opcode.toString(), name)) {
+            return opcode;
+        }
+    }
+    return null;
+}
+
+// Helper function to get push data length
+pub fn getPushDataLength(opcode: u8) ?u8 {
```

Comment:
Change the argument to `opcode: Opcode` otherwise people will be able to pass invalid u8 like `2` or `7` and you if branch will still accept them.